common/src/Model/ParallelTexCoordSystem.cpp

```diff
@@ -25,13 +25,14 @@
 namespace TrenchBroom {
     namespace Model {
         ParallelTexCoordSystemSnapshot::ParallelTexCoordSystemSnapshot(ParallelTexCoordSystem* coordSystem) :
-        m_coordSystem(coordSystem),
-        m_xAxis(m_coordSystem->xAxis()),
-        m_yAxis(m_coordSystem->yAxis()) {}
+        m_xAxis(coordSystem->xAxis()),
+        m_yAxis(coordSystem->yAxis()) {}
         
-        void ParallelTexCoordSystemSnapshot::doRestore() {
-            m_coordSystem->m_xAxis = m_xAxis;
-            m_coordSystem->m_yAxis = m_yAxis;
+        void ParallelTexCoordSystemSnapshot::doRestore(TexCoordSystem* coordSystem) const {
+            ParallelTexCoordSystem *parallelCoordSystem = dynamic_cast<ParallelTexCoordSystem*>(coordSystem);
```

Comment:
Mmm, good point, I'll rework it with double dispatch to avoid the cast.

I think dynamic_cast should be used when casting across class hierarchies - otherwise you're punching a hole in the type system, and not checking if it was successful or not.